packages/app/hooks/use-persist-form.ts

```diff
@@ -38,87 +37,133 @@ export const usePersistForm = (
+            } else {
+              const value = values[key];
+              dataRestored[key] = value;
+              setValue(key, value, {
+                shouldValidate: validate,
+                shouldDirty: dirty,
+                shouldTouch: touch,
+              });
             }
-          } else {
-            dataRestored[key] = values[key] || defaultValues?.[key];
-            setValue(key, values[key], {
+          }
+        }
+      } else if (defaultValues) {
+        for (let key in defaultValues) {
+          const shouldSet = !exclude.includes(key);
+          if (shouldSet) {
+            const value = defaultValues?.[key];
+            dataRestored[key] = value;
+            setValue(key, value, {
               shouldValidate: validate,
               shouldDirty: dirty,
               shouldTouch: touch,
             });
           }
         }
-      });
-
-      if (onDataRestored) {
-        onDataRestored(dataRestored);
       }
-    } else if (defaultValues) {
-      Object.keys(defaultValues).forEach((key) => {
-        const shouldSet = !exclude.includes(key);
-        if (shouldSet) {
-          setValue(key, defaultValues[key], {
-            shouldValidate: validate,
-            shouldDirty: dirty,
-            shouldTouch: touch,
-          });
-        }
-      });
+
+      onDataRestored?.(dataRestored);
     }
+    restoreForm();
+
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [name, onDataRestored, setValue]);
 
   useEffect(() => {
-    let stringValues: any = {};
-    for (let key in watchedValues) {
-      if (!exclude.includes(key)) {
-        if (watchedValues[key] instanceof File) {
-          fileStorage.saveFile(watchedValues[key], key);
-          stringValues[key] = "instanceof File";
-        } else {
-          stringValues[key] = watchedValues[key];
+    function persistFormValues() {
+      let stringValues: any = {};
+      for (let key in watchedValues) {
+        if (!exclude.includes(key)) {
+          if (watchedValues[key] instanceof File) {
+            fileStorage.saveFile(watchedValues[key], key);
+            stringValues[key] = "instanceof File";
+          } else {
+            stringValues[key] = watchedValues[key];
+          }
         }
       }
-    }
 
-    if (Object.keys(stringValues).length) {
-      if (timeout !== undefined) {
-        stringValues._timestamp = Date.now();
+      if (Object.keys(stringValues).length) {
+        if (timeout !== undefined) {
+          stringValues._timestamp = Date.now();
+        }
+        store.set(name, JSON.stringify(stringValues));
       }
-      store.set(name, JSON.stringify(stringValues));
     }
-  }, [watchedValues, timeout, exclude, fileStorage, name]);
+
+    if (persistDebounceTimeout.current) {
+      clearTimeout(persistDebounceTimeout.current);
+    }
+    persistDebounceTimeout.current = setTimeout(() => {
+      if (Object.keys(restoringFiles).length === 0) persistFormValues();
```

Comment:
we only start storing values in localStorage after restoringFiles state becomes empty (can take max [2 seconds](https://github.com/showtime-xyz/showtime-frontend/pull/1882/files#diff-e984ccef2e92f4513764a35a6c62974e4997d0379850b0cb20b1312ee4113cc0R87) in worst case.)